src/corehost/common/pal.windows.cpp

```diff
@@ -227,7 +240,44 @@ bool pal::get_default_installation_dir(pal::string_t* recv)
     }
 
     append_path(recv, _X("dotnet"));
+
+    return true;
+}
+
+bool pal::get_sdk_self_registered_dir(pal::string_t* recv)
+{
+#if _TARGET_ARM_
+    //  Self-registered SDK installation directory is not supported for win-arm
+    return false;
+#else
+    recv->clear();
+
+    DWORD size = 0;
+    const HKEY hkey = HKEY_LOCAL_MACHINE;
+    // The registry search occurs in the 32-bit registry in all cases.
+    const DWORD flags = RRF_RT_REG_SZ | RRF_SUBKEY_WOW6432KEY;
+    pal::string_t sub_key = pal::string_t(_X("SOFTWARE\\dotnet\\Setup\\InstalledVersions\\")) + get_arch() + pal::string_t(_X("\\sdk"));
+    pal::char_t* value = _X("InstallLocation");
+
+    // Determine the size of the buffer
+    LONG result = ::RegGetValueW(hkey, sub_key.c_str(), value, flags, nullptr, nullptr, &size);
+    if (result != ERROR_SUCCESS || size == 0)
+    {
+        return false;
+    }
+
+    // Get the key's value
+    std::vector<wchar_t> buffer(size/sizeof(wchar_t));
```

Comment:
Should we be using `pal::char_t` over `wchar_t` throughout? We're relying on them being the same on Windows in passing sub_key.c_str() to RegGetValueW anyway, correct?

Reply:
Good point - we should be consistent...